theseus_gui/src/store/instancesStore.js

```diff
@@ -0,0 +1,337 @@
+        }
+        if (this.environments.server === true) {
+          facets +=
+            '["server_side:optional"],["server_side:required"],["client_side:optional"],["client_side:unsupported"],'
+        }
+
+        if (this.openSource === true) facets += '["open_source:true"],'
+
+        facets = facets.slice(0, facets.length - 1)
+        facets += ']'
+      }
+
+      let indexSort
+      switch (this.filter) {
+        case 'Download count':
+          indexSort = 'downloads'
+          break
+        case 'Follow count':
+          indexSort = 'follows'
+          break
+        case 'Recently published':
+          indexSort = 'newest'
+          break
+        case 'Recently updated':
+          indexSort = 'updated'
+          break
+        default:
+          indexSort = 'relevance'
+      }
+
+      const response = await ofetch(
+        `https://api.modrinth.com/v2/search?query=${this.searchInput || ''}&limit=${
+          this.limit
+        }&offset=${this.offset || 0}${facets || ''}&index=${indexSort}`
+      )
+      this.instances = [...response.hits]
+      this.totalHits = response.total_hits
+      this.offset = response.offset
+      this.pageCount = Math.ceil(this.totalHits / 10)
+    },
+    setSearchInput(newInput) {
+      this.searchInput = newInput
+    },
+    setCurrentPage(newPage) {
+      this.currentPage = newPage
+
+      if (newPage === 1) this.offset = 0
+      else this.offset = this.currentPage * 10 - 10
+    },
+    setFilter(newFilter) {
+      this.filter = newFilter
+    },
+    toggleCategory(cat) {
+      this.categories[cat] = !this.categories[cat]
+    },
+    toggleLoader(loader) {
+      this.loaders[loader] = !this.loaders[loader]
+    },
+    toggleEnv(env) {
+      this.environments[env] = !this.environments[env]
+    },
+    setVersions(versions) {
+      this.activeVersions = versions
+    },
+    setLimit(newLimit) {
+      this.limit = newLimit
+    },
+    resetFilters() {
+      Object.keys(this.categories).forEach((cat) => {
+        this.categories[cat].enabled = false
+      })
+      Object.keys(this.loaders).forEach((loader) => {
+        this.loaders[loader].enabled = false
+      })
+      Object.keys(this.environments).forEach((env) => {
+        this.environments[env] = false
+      })
+      this.activeVersions = []
+    },
+  },
+  getters: {
+    getFilteredInstances: (state) => {
+      const filteredInstances = state.fakeInstances.filter((i) => {
+        // When time comes, do more advanced client-side filtering here if wise
+        const normalizedInstanceName = i.name?.toLowerCase()
+        if (normalizedInstanceName.includes(state.filter.toLowerCase())) return i
+      })
+
+      if (filteredInstances && filteredInstances.length > 0) return filteredInstances
+
+      return state.fakeInstances
+    },
+    getCategoriesByInstanceId: (state) => {
```

Comment:
what does this func do?

Reply:
Whatever id is passed in, it gets that project's categories and finds the corresponding icon svgs for them. Those svgs are then passed into the ProjectCard as props.